xml/issue3371.xml

```diff
@@ -0,0 +1,57 @@
+<?xml version='1.0' encoding='utf-8' standalone='no'?>
+<!DOCTYPE issue SYSTEM "lwg-issue.dtd">
+
+<issue num="3371" status="New">
+<title><tt>visit_format_arg</tt> and <tt>make_format_args</tt> are not hidden friends</title>
+<section><sref ref="[format.arg]"/></section>
+<submitter>Tim Song</submitter>
+<date>16 Jan 2020</date>
+<priority>99</priority>
+
+<discussion>
+<p>
+After <a href="https://wg21.link/P1965R0">P1965R0</a>, friend function and function template
+declarations always introduce hidden friends under the new new blanket wording in
+<sref ref="[hidden.friends]"/>. However, <sref ref="[format.arg]"/> contains
+"exposition only" friend declarations of <tt>visit_format_arg</tt> and <tt>make_format_args</tt>,
+and those are not intended to be hidden. The only reason to have these declarations in the first
+place is because these function templates are specified using the exposition-only private data
+members of <tt>basic_format_arg</tt>, but that's unnecessary &mdash; for example,
+<tt>shared_ptr</tt>'s constructors aren not exposition-only friends <tt>enable_shared_from_this</tt>,
+even though the former are shown as assigning to the latter's exposition-only <tt>weak_this</tt>
+private data member (see <sref ref="[util.smartptr.shared.const]"/>p1).
```

Comment:
Of the _`iterator`_ s in [ranges], `element_view`'s the only one to befriend its counterpart _`iterator`_`<!Const>`. Do you think the same logic applies there, and thus should be removed?

Reply:
> `elements_view`'s the only one to befriend its counterpart

Ouch, yes - that's implementation detail that leaked into the wording. I'll submit a separate issue to fix if Tim doesn't want to add that change here.

Reply:
We have random friends at various places in the standard, and it's normative harmless, so I'd prefer to do that separately.

Reply:
@timsong-cpp: Could you shortly reply to Casey's comment so that I know what your recommended direction will be?

Reply:
@Dani-Hub I'd like to do that separately; the "hidden friends" global wording doesn't affect it, since `iterator<!Const>` is not a function or function template.